test/javascript/erlang/erlang_test.mjs

```
@@ -6662,6 +6662,147 @@ describe("Erlang", () => {
+      const expected = Type.integer(-9_007_199_254_740_991n);
+
+      assert.deepStrictEqual(result, expected);
+    });
+
+    it("handles large float that loses precision", () => {
+      // This demonstrates that float representation limits apply before rounding
+      // 36_028_797_018_963_969.0 cannot be represented exactly as a float
+      // It's stored as 36_028_797_018_963_968.0
+      // eslint-disable-next-line no-loss-of-precision
+      const result = testedFun(Type.float(36_028_797_018_963_969.0));
+      const expected = Type.integer(36_028_797_018_963_968n);
```

Biome (2.3.13)

[error] This number literal will lose precision at runtime.

The value at runtime will be 36028797018963970

(lint/correctness/noPrecisionLoss)